src/main/java/org/kairosdb/core/http/rest/MetricsResource.java

```diff
@@ -523,6 +528,8 @@ public Response runQuery(String json, String remoteAddr) throws Exception
 				ThreadReporter.addTag("metric_name", query.getName());
 				ThreadReporter.addTag("query_index", String.valueOf(queryCount));
 
+				sourceIPLog.append(query.getName() + ", ");
+
```

Comment:
@guildencrantz @ehatch90  After I created this pull request I had the thought that this would allow for duplicates in the list of metrics queried, as people could query for the same metric many times in one query. I think changing this to add the metric names to a HashSet and then adding the string of that to the log entry might be better. What are your thoughts?

Reply:
Generically speaking: Agreed. However, would it be useful to know if they're hitting the same one multiple times? Or do we have the whole query elsewhere that we can tie back to? If we've got it else where then yes, let's keep this tight. If not maybe leave it?

Reply:
We don't have the whole query unless the time it takes is longer than the configured threshold (60s currently I think), and then it would be pushed to cassandra as a datapoint in the kairosdb.log.query.json metric.

Reply:
@guildencrantz @ehatch90 Okay, let's leave this as it is (with the possible duplicates).